### PR TITLE
[sycl-post-link] Fix the order of spec constants IDs in device image properties

### PR DESCRIPTION
**Pull request by vmaksimo**

### CODE REVIEW

**Review by AlexeySachkov — APPROVED** [review]
LGTM

@dm-vodopyanov: FYI

**Review by AlexeySachkov — CHANGES_REQUESTED** [review]
But we should probably add a test

**Review by AlexeySachkov — APPROVED** [review]

**Comment by vmaksimo** [comment]
Ping @rbegam

**Review by rbegam — APPROVED** [review]
LGTM

File: llvm/tools/sycl-post-link/SpecConstants.h [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 #pragma once
 
+#include "llvm/ADT/MapVector.h"
```
Comment (bader) [inline llvm/tools/sycl-post-link/SpecConstants.h:18]: Please, remove `#include <map>` from line 22.

**Comment by bader** [comment]
@vmaksimo, this change introduced a regression to check-llvm tests. https://github.com/intel/llvm/runs/2604002244
Please, fix ASAP.

**Comment by AlexeySachkov** [comment]
> @vmaksimo This patch also introduced llvm-dis as dependency for check-sycl target which is not defined in CMake files.
> Could you please remove that dependency or define it explicitly in https://github.com/intel/llvm/blob/sycl/sycl/test/CMakeLists.txt#L30
> 
> The problem is not exposed in CI because this dependency is built on preceding steps in test jobs.

I suggest that we remove the dependency, because it should be possible to write the test using text LLVM IR representation only

**Comment by vmaksimo** [comment]
> I suggest that we remove the dependency, because it should be possible to write the test using text LLVM IR representation only

`llvm-spirv` does not accept LLVM IR as an input - so we should use either `llvm-dis` (as it's done now) or `llvm-as` to run `llvm-spirv`.

**Comment by AlexeySachkov** [comment]
> > I suggest that we remove the dependency, because it should be possible to write the test using text LLVM IR representation only
> 
> `llvm-spirv` does not accept LLVM IR as an input - so we should use either `llvm-dis` (as it's done now) or `llvm-as` to run `llvm-spirv`.

Since we do not check generated SPIR-V I think we can avoid launching `llvm-spirv` in that test.

**Comment by vmaksimo** [comment]
Removed dependency in https://github.com/intel/llvm/pull/3821

**Pull request merged**